src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -1989,6 +1989,11 @@
    options <- options("repos", "download.file.method", "download.file.extra", "HTTPUserAgent")
    if (identical(options[["download.file.method"]], "curl"))
       options[["download.file.extra"]] <- .rs.downloadFileExtraWithCurlArgs()
+
+   # if there is no CRAN mirror specified in the R session, read it from preferences
+   if (is.na(options[["repos"]]["CRAN"])) {
```

Comment:
I believe this relies on the `repos` option being a character vector, but it's technically legal for it to be a list; e.g. some users might do something like:

```
options(repos = list(RSPM = "https://packagemanager.rstudio.com/cran/latest"))
```

Reply:
Hrmmm, that's a good point. How would you feel about avoiding messing with `repos` unless it isn't set at all? (i.e. any non-NULL option would be passed through, but if NULL we would fill in the CRAN repo from user prefs?)

Reply:
That sounds good to me; alternatively, any length-zero option just to catch cases where it's an empty list or vector as well.

Reply:
Done!